packages/legend-extension-dsl-diagram/src/helpers/DiagramHelper.ts

```diff
@@ -51,14 +52,14 @@ export const cleanUpDeadReferencesInDiagram = (
         .includes(propertyView.property.value.name),
   );
   propertyViewsToRemove.forEach((propertyView) =>
-    diagram.deletePropertyView(propertyView),
+    deleteEntry(diagram.propertyViews, propertyView),
```

Comment:
These are called in graph methods, they could be called on an observable graph. I'm not sure if this is safe to do. Could you test by deleting some classes on a diagram?